synthtool/metadata.py

```diff
@@ -238,3 +242,52 @@ def _clear_local_paths(metadata):
         if source.HasField("git"):
             git_source = source.git
             git_source.ClearField("local_path")
+
+
+def _add_self_git_source():
+    """Adds current working directory as a git source.
+
+    Returns:
+        The number of git sources added to metadata.
+    """
+    # Use the repository's root directory name as the name.
+    return _add_git_source_from_directory(".", os.getcwd())
+
+
+def _add_git_source_from_directory(name: str, dir_path: str):
+    """Adds the git repo containing the directory as a git source.
+
+    Returns:
+        The number of git sources added to metadata.
+    """
+    completed_process = subprocess.run(
+        ["git", "-C", dir_path, "status"], universal_newlines=True
+    )
+    if completed_process.returncode:
+        log.warning("%s is not directory in a git repo.", dir_path)
+        return 0
+    completed_process = subprocess.run(
+        ["git", "-C", dir_path, "remote", "get-url", "origin"],
+        stdout=subprocess.PIPE,
+        universal_newlines=True,
+    )
+    url = completed_process.stdout.strip()
+    completed_process = subprocess.run(
+        ["git", "-C", dir_path, "log", "--no-decorate", "-1", "--pretty=format:%H"],
+        stdout=subprocess.PIPE,
+        universal_newlines=True,
+    )
+    latest_sha = completed_process.stdout.strip()
+    add_git_source(name=name, remote=url, sha=latest_sha)
+    return 1
+
+
+def _add_synthtool_git_source():
+    """Adds synthtool's repo as a git source.
+
+    Returns:
+        The number of git sources added to metadata.
+    """
+    source_path = inspect.getfile(synthtool)
```

Comment:
In the case of pypi, no, the synthtool source won't be recorded in the metadata.  Autosynth is always uses a git repo, and that's the case I have in mind.  How common is it for DPEs to run synthtool manually and commit the results?

Reply:
I usually run it locally (albeit from a git clone of synthtool) when we're modifying `synth.py` (often for a new service version), but that is usually because I don't want to wait for the next autosynth run to merge the changes as well.

We haven't actually released synthtool for awhile so maybe this is not a concern. Although, we should probably stop showing the pypi install instructions if we don't plan on pushing releases to pypi.

Reply:
I'll take a survey during our next team meeting to see how people are using synthtool.